requests/adapters.py

```diff
@@ -272,13 +272,10 @@ def request_url(self, request, proxies):
         :class:`HTTPAdapter <requests.adapters.HTTPAdapter>`.
 
         :param request: The :class:`PreparedRequest <PreparedRequest>` being sent.
-        :param proxies: A dictionary of schemes to proxy URLs.
+        :param proxies: A dictionary of schemes or schemes and hosts to proxy URLs.
         """
-        proxies = proxies or {}
-        scheme = urlparse(request.url).scheme
-        proxy = proxies.get(scheme)
-
-        if proxy and scheme != 'https':
+        proxy = select_proxy(request.url, proxies)
+        if proxy and not request.url.lower().startswith('https'):
```

Comment:
This is a regressing, breaking change. It's possible that `request.url` may begin with `https` but not actually be a HTTPS request. Please keep the urlparse logic in place. =)